### PR TITLE
Research indexing and replay performance

### LINKED ISSUE #1: Research index/replay performance
To help ground our approach and plan the work cycle it would be useful to have performance statistics on:

* time to re-index WARC content
* time to replay content using one pywb collection (one index)
* time to replay content using multiple pywb collections (multiple indexes)
* time to replay content with OutbackCDX

We should aim to index ~1TB of WARC content using pywb and something like ~50 collections. Based on this it should be possible to calculate a MB/sec rate to use for estimating how long the complete reindex should take.

Once finished create a single pywb index using a [sort command](https://github.com/sul-dlss/was_robot_suite/blob/main/lib/was_crawl_dissemination/cdx_merge_sort_publish_service.rb#L21). Add this index (or the separate ones) to OutbackCDX. Run a test using [wrk](https://github.com/wg/wrk) (or something like it) to see how the three indexing approaches perform.

### PR DESCRIPTION
**Pull request by edsu**
I've added the results of indexing and querying performance testing to #1. This PR also includes some changes to the Docker setup to reflect how the was-pywb-stage/prod have been set up. However it will need more adjustment to use Apache instead of Nginx (which is what ops decided to use).

Closes #1

### CODE REVIEW

**Comment by lwrubel** [comment]
Running cdxj-indexer on a single 1.1TB collection took 447 minutes (7.45 hours) and produced a 18GB index. 

Index file is on was-dev at
`/data/lwrubel/sr233xh9483-index.cdxj`

At that rate, re-indexing 46TB of WARCs would be roughly 13 days. Will be good to cross check with the script and different collections.

**Comment by edsu** [comment]
## Indexing Performance

Indexing of all the WARC content took ~62 hours with 7 concurrent processes. So we will be able to take several passes at reindexing during the work-cycle if necessary. We may want to reindex if we choose to continue using the [path index](https://pywb.readthedocs.io/en/latest/manual/owb-to-pywb-config.html#path-index-configuration) configuration instead of setting the full path in the CDXJ file as I did this time with `--dir-root`. The druid tree path gets repeated quite a bit in these CDXJ files, so it will cut down on their size to use the path index.

## Replay Performance

The following stats show how long it took to read a response from the CDX endpoint for one URL when using multiple collections vs one collection. The *cold* time is after the operating system page cache was flushed, and *warm* was the same call repeated again (after caching). This was only testing with an index containing a total of 91,781,768 lines and 40GB in size. We can expect the full index to be around 240GB.

### Replay of a URL from "all"

This tested querying the "all" virtual collection with 20 collections.

```
$ time curl 'http://localhost:8080/all/cdx?url=http://69.63.136.213/documents/Comments%20--%20Final.pdf'
```

**cold**
real    0m11.211s
user    0m0.010s
sys     0m0.026s

**warm**
real    0m0.866s
user    0m0.006s
sys     0m0.012s

### Replay from one big index

```
$ time curl 'http://localhost:8080/one/cdx?url=http://69.63.136.213/documents
```

**cold**
real    0m1.931s
user    0m0.011s
sys     0m0.022s

**warm**
real    0m0.020s
user    0m0.000s
sys     0m0.009s

Based on this I think we may want to consider starting with a one index approach, mirroring what we are currently doing with openwayback.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # was-pywb
 
-This repository contains an experimental Docker configuration for a SUL [pywb] instance. It uses [nginx] as a reverse proxy in front of pywb, which is configured to use [OutbackCDX] for indexing. 
+This repository contains a Docker configuration for a SUL [pywb] instance for use in development. It uses [nginx] as a reverse proxy in front of pywb, which is configured to use [OutbackCDX] for indexing.  It uses the default configuration of the stage and prod instances.
```
Comment (lwrubel) [inline README.md:3]: Remove OutbackCDX

File: bin/index.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+# This program will walk through WARC files and generate CDXJ files for them.
+
+require 'logger'
+require 'optparse'
+require 'pathname'
+require 'parallel'
+
+WARC_DIR = '/web-archiving-stacks/data/collections'
+CDX_DIR = '/data/edsu/indexes'
```
[the remark this answers is not in the document]
Comment (edsu) [reply bin/index.rb:12]: Yes, was-dev was setup to have /web-archiving-stacks be read-only. Look for this to change as we start to use it for real in production.

File: .gitignore [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+/pkg/
+/spec/reports/
+/tmp/
+/.DS_Store
 
-*.log
+# rspec failure tracking
+.rspec_status
+
+Gemfile.lock
```
[the remark this answers is not in the document]
Comment (edsu) [reply .gitignore]: Thanks, I just pushed up a change with that fixed.

**Comment by edsu** [comment]
I removed web-archiving-stacks from the repository so that it doesn't get pulled into the stage, qa and prod environments. Users can copy the data in manually using `docker cp`. Thanks for pointing this out!

**Pull request merged**